src/py42/clients/settings/device_settings.py

```diff
@@ -0,0 +1,349 @@
+        location=["settings", "serviceBackupConfig", "severeEmailEnabled"],
+        get_converter=str_to_bool,
+        set_converter=bool_to_str,
+    )
+    warning_alert_days = SettingProperty(
+        name="warning_alert_days",
+        location=["settings", "serviceBackupConfig", "minutesUntilWarning"],
+        get_converter=minutes_to_days,
+        set_converter=days_to_minutes,
+    )
+    critical_alert_days = SettingProperty(
+        name="critical_alert_days",
+        location=["settings", "serviceBackupConfig", "minutesUntilSevere"],
+        get_converter=minutes_to_days,
+        set_converter=days_to_minutes,
+    )
+    backup_status_email_enabled = SettingProperty(
+        name="backup_status_email_enabled",
+        location=["settings", "serviceBackupConfig", "backupStatusEmailEnabled"],
+        get_converter=str_to_bool,
+        set_converter=bool_to_str,
+    )
+    backup_status_email_frequency_days = SettingProperty(
+        name="backup_status_email_frequency_days",
+        location=["settings", "serviceBackupConfig", "backupStatusEmailFreqInMinutes"],
+        get_converter=minutes_to_days,
+        set_converter=days_to_minutes,
+    )
+
+    def __repr__(self):
+        return "<DeviceSettingsDefaults: org_id: {}>".format(self._org_settings.org_id)
+
+
+class DeviceSettings(DeviceSettingsDefaults):
+    """Class used to manage an individual device's settings."""
+
+    def __init__(self, device_dict):
+        self.changes = {}
+        self.data = device_dict
+        self._destinations = device_dict["availableDestinations"]
+        self.backup_sets = [
+            BackupSet(self, self.changes, backup_set)
+            for backup_set in self.data["settings"]["serviceBackupConfig"][
+                "backupConfig"
+            ]["backupSets"]
+        ]
+
+    @property
+    def computer_id(self):
+        return self.data["computerId"]
+
+    @property
+    def guid(self):
+        return self.data["guid"]
+
+    @property
+    def org_id(self):
+        return self.data["orgId"]
+
+    @property
+    def user_id(self):
+        return self.data["userId"]
+
+    @property
+    def version(self):
+        return self.data["version"]
+
+    name = SettingProperty(name="name", location=["name"])
+    external_reference = SettingProperty(
+        name="external_reference", location=["computerExtRef"]
+    )
+    notes = SettingProperty(name="notes", location=["notes"])
+
+    def __repr__(self):
+        return "<DeviceSettings: guid: {}, name: {}>".format(
+            self.data["guid"], self.data["name"]
+        )
+
+
+class BackupSet(UserDict, object):
+    """Helper class for managing device backup sets and Org device default backup sets."""
+
+    def __init__(self, settings_manager, changes_dict, backup_set_dict):
+        self._manager = settings_manager
+        self._changes = changes_dict
+        self.data = backup_set_dict
+        includes, excludes = self._extract_file_selection_lists()
```

Comment:
there are a lot of methods in the class... It'd be easier to review if the internal ones were ordered in the order of the usage so I can follow along easier

Reply:
Yeah, I'm working on a bit of cleanup on that front and adding some docstrings to make it clear why these exist now.